testing/integration/endpoint_security_test.go

```diff
@@ -894,3 +901,536 @@ func TestForceInstallOverProtectedPolicy(t *testing.T) {
+
+				assert.Equal(t, proxyCLI.URL, got.Fleet.ProxyURL)
+				assert.Equal(t, mtlsCLI.clientCertPath, got.Fleet.Ssl.Certificate)
+				assert.Equal(t, mtlsCLI.clientCertKeyEncPath, got.Fleet.Ssl.Key)
+				assert.Equal(t, mtlsCLI.clientCertKeyPassPath, got.Fleet.Ssl.KeyPassphrasePath)
+			},
+		},
+		{
+			Name:            "proxy-from-cli-and-passphrase-protected-cert-key-proxy-from-policy-one-way-TLS",
+			URL:             fleethostWrong.String(),
+			EnrollmentToken: enrollmentTokenOneWayProxyResp.APIKey,
+			ProxyURL:        proxyCLI.URL,
+
+			CertificateAuthorities: []string{mtlsCLI.proxyCAPath},
+			Certificate:            mtlsCLI.clientCertPath,
+			Key:                    mtlsCLI.clientCertKeyEncPath,
+			KeyPassphrase:          mtlsCLI.clientCertKeyPassPath,
+			assertInspect: func(t *testing.T, f *atesting.Fixture) {
+				// wait for the agent to apply the policy coming from fleet-server
+				buff := &strings.Builder{}
+				assert.Eventuallyf(t, func() bool {
+					buff.Reset()
+
+					got, err := f.ExecInspect(ctx)
+					if err != nil {
+						buff.WriteString(fmt.Sprintf("error running inspect cmd: %v", err))
+						return false
+					}
+
+					return proxyPolicyOneWayTLS.URL == got.Fleet.ProxyURL
+				}, time.Minute, time.Second, "inspect never showed proxy from policy: %s", buff)
+
+				t.Skip("remove skip once https://github.com/elastic/elastic-agent/issues/5888 is fixed")
+				got, err := f.ExecInspect(ctx)
+				require.NoErrorf(t, err, "error running inspect cmd")
+
+				assert.Equal(t, proxyPolicyOneWayTLS.URL, got.Fleet.ProxyURL)
+				assert.Equal(t, []string{oneWayTLSPolicy.proxyCAPath}, got.Fleet.Ssl.CertificateAuthorities)
+				assert.Empty(t, got.Fleet.Ssl.Certificate, "client certificate isn't present in the proxy from the policy")
+				assert.Empty(t, got.Fleet.Ssl.Key, "client certificate key isn't present in the proxy from the policy")
+				assert.Empty(t, got.Fleet.Ssl.KeyPassphrasePath, "client certificate key passphrase isn't present in the proxy from the policy")
+			},
+		},
+		{
+			Name:            "proxy-from-cli-and-policy-both-with-plain-cert-key",
+			URL:             fleethostWrong.String(),
+			EnrollmentToken: enrollmentTokenmTLSProxyResp.APIKey,
+			ProxyURL:        proxyCLI.URL,
+
+			CertificateAuthorities: []string{mtlsCLI.proxyCAPath},
+			Certificate:            mtlsCLI.clientCertPath,
+			Key:                    mtlsCLI.clientCertKeyPath,
+
+			assertInspect: func(t *testing.T, f *atesting.Fixture) {
+				// wait for the agent to apply the policy coming from fleet-server
+				buff := &strings.Builder{}
+				assert.Eventuallyf(t, func() bool {
+					buff.Reset()
+
+					got, err := f.ExecInspect(ctx)
+					if err != nil {
+						buff.WriteString(fmt.Sprintf("error running inspect cmd: %v", err))
+						return false
+					}
+
+					return proxyPolicymTLS.URL == got.Fleet.ProxyURL
+				}, time.Minute, time.Second, "inspect never showed proxy from policy: %s", buff)
```

Comment:
`buff` is really the last error encountered in the assert.Eventually, maybe we can define and treat it as such:



```suggestion
				// feel free to change the variable name
				var buff error
				assert.Eventuallyf(t, func() bool {

					got, err := f.ExecInspect(ctx)
					if err != nil {
						buff = fmt.Errorf("error running inspect cmd: %w", err)
						return false
					}
					// (optional) reset the error
					buff = nil
					return proxyPolicymTLS.URL == got.Fleet.ProxyURL
				}, time.Minute, time.Second, "inspect never showed proxy from policy: %s", buff)
```